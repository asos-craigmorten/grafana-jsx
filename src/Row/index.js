import PropTypes from "prop-types";
import Panel from "../Panel";
import Panels from "../Panels";

const Row = ({
  children,
  collapsed = false,
  datasource,
  options,
  title,
  y,
  ...props
}) => {
  if (collapsed) {
    return (
      <Panel
        collapsed={collapsed}
        datasource={datasource}
        height={1}
        options={options}
        title={title}
        type={"row"}
        width={24}
        x={0}
        y={y}
        {...props}
      >
        <Panels>{children}</Panels>
      </Panel>
    );
  }

  return (
    <>
      <Panel
        collapsed={collapsed}
        datasource={datasource}
        height={1}
        options={options}
        title={title}
        type={"row"}
        width={24}
        x={0}
        y={y}
        {...props}
      >
        <Panels />
      </Panel>
      {children}
    </>
  );
};

Row.PropTypes = {
  children: PropTypes.arrayOf(PropTypes.object),
  collapsed: PropTypes.bool,
  datasource: PropTypes.string,
  title: PropTypes.string,
  y: PropTypes.number.isRequired,
  options: PropTypes.object,
};

export default Row;
