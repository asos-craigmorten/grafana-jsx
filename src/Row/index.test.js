import Row from "./";

jest.mock("../Panels", () => ({ children }) => ({ "test-panels": children }));

const requiredProps = {
  y: 0,
};

const withAllProps = {
  ...requiredProps,
  collapsed: true,
  datasource: "test-datasource",
  title: "test-title",
  options: { "test-option-key": "test-option-value" },
};

const TestArray = ({ children }) => children;

describe("Row", () => {
  describe("when used in isolation", () => {
    describe("when passed no Panel children", () => {
      describe("when passed required props", () => {
        it("should render a default row with required props", () => {
          expect(<Row {...requiredProps} />).toMatchSnapshot();
        });
      });

      describe("when passed all props", () => {
        it("should render a custom row", () => {
          expect(<Row {...withAllProps} />).toMatchSnapshot();
        });
      });

      describe("when passed custom props", () => {
        it("should render a custom row with the custom props", () => {
          expect(
            <Row {...withAllProps} test-row-prop={Symbol("test-row-value")} />
          ).toMatchSnapshot();
        });
      });
    });

    describe("when passed Panel children", () => {
      describe("and the panel is collapsed", () => {
        describe("when passed required props", () => {
          it("should render a default row with required props", () => {
            expect(
              <Row {...requiredProps} collapsed={true}>
                <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
              </Row>
            ).toMatchSnapshot();
          });
        });

        describe("when passed all props", () => {
          it("should render a custom row", () => {
            expect(
              <Row {...withAllProps} collapsed={true}>
                <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
              </Row>
            ).toMatchSnapshot();
          });
        });

        describe("when passed custom props", () => {
          it("should render a custom row with the custom props", () => {
            expect(
              <Row
                {...withAllProps}
                collapsed={true}
                test-row-prop={Symbol("test-row-value")}
              >
                <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
              </Row>
            ).toMatchSnapshot();
          });
        });
      });

      describe("and the panel is not collapsed", () => {
        describe("when passed required props", () => {
          it("should render a default row with required props", () => {
            expect(
              <Row {...requiredProps} collapsed={false}>
                <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
              </Row>
            ).toMatchSnapshot();
          });
        });

        describe("when passed all props", () => {
          it("should render a custom row", () => {
            expect(
              <Row {...withAllProps} collapsed={false}>
                <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
              </Row>
            ).toMatchSnapshot();
          });
        });

        describe("when passed custom props", () => {
          it("should render a custom row with the custom props", () => {
            expect(
              <Row
                {...withAllProps}
                collapsed={false}
                test-row-prop={Symbol("test-row-value")}
              >
                <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
              </Row>
            ).toMatchSnapshot();
          });
        });
      });
    });
  });

  describe("when used within another object", () => {
    describe("when passed no Panel children", () => {
      describe("when passed required props", () => {
        it("should render a default row with required props", () => {
          expect(
            <test>
              <Row {...requiredProps} />
            </test>
          ).toMatchSnapshot();
        });
      });

      describe("when passed all props", () => {
        it("should render a custom row", () => {
          expect(
            <test>
              <Row {...withAllProps} />
            </test>
          ).toMatchSnapshot();
        });
      });

      describe("when passed custom props", () => {
        it("should render a custom row with the custom props", () => {
          expect(
            <test>
              <Row {...withAllProps} test-row-prop={Symbol("test-row-value")} />
            </test>
          ).toMatchSnapshot();
        });
      });
    });

    describe("when passed Panel children", () => {
      describe("and the panel is collapsed", () => {
        describe("when passed required props", () => {
          it("should render a default row with required props", () => {
            expect(
              <test>
                <Row {...requiredProps} collapsed={true}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </test>
            ).toMatchSnapshot();
          });
        });

        describe("when passed all props", () => {
          it("should render a custom row", () => {
            expect(
              <test>
                <Row {...withAllProps} collapsed={true}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </test>
            ).toMatchSnapshot();
          });
        });

        describe("when passed custom props", () => {
          it("should render a custom row with the custom props", () => {
            expect(
              <test>
                <Row
                  {...withAllProps}
                  collapsed={true}
                  test-row-prop={Symbol("test-row-value")}
                >
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </test>
            ).toMatchSnapshot();
          });
        });
      });

      describe("and the panel is not collapsed", () => {
        describe("when passed required props", () => {
          it("should render a default row with required props", () => {
            expect(
              <test>
                <Row {...requiredProps} collapsed={false}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </test>
            ).toMatchSnapshot();
          });
        });

        describe("when passed all props", () => {
          it("should render a custom row", () => {
            expect(
              <test>
                <Row {...withAllProps} collapsed={false}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </test>
            ).toMatchSnapshot();
          });
        });

        describe("when passed custom props", () => {
          it("should render a custom row with the custom props", () => {
            expect(
              <test>
                <Row
                  {...withAllProps}
                  collapsed={false}
                  test-row-prop={Symbol("test-row-value")}
                >
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </test>
            ).toMatchSnapshot();
          });
        });
      });
    });
  });

  describe("when used within another object's array", () => {
    describe("when passed no Panel children", () => {
      describe("when passed required props", () => {
        it("should render a default row with required props", () => {
          expect(
            <TestArray>
              <Row {...requiredProps} />
            </TestArray>
          ).toMatchSnapshot();
        });
      });

      describe("when passed all props", () => {
        it("should render a custom row", () => {
          expect(
            <TestArray>
              <Row {...withAllProps} />
            </TestArray>
          ).toMatchSnapshot();
        });
      });

      describe("when passed custom props", () => {
        it("should render a custom row with the custom props", () => {
          expect(
            <TestArray>
              <Row {...withAllProps} test-row-prop={Symbol("test-row-value")} />
            </TestArray>
          ).toMatchSnapshot();
        });
      });
    });

    describe("when passed Panel children", () => {
      describe("and the panel is collapsed", () => {
        describe("when passed required props", () => {
          it("should render a default row with required props", () => {
            expect(
              <TestArray>
                <Row {...requiredProps} collapsed={true}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </TestArray>
            ).toMatchSnapshot();
          });
        });

        describe("when passed all props", () => {
          it("should render a custom row", () => {
            expect(
              <TestArray>
                <Row {...withAllProps} collapsed={true}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </TestArray>
            ).toMatchSnapshot();
          });
        });

        describe("when passed custom props", () => {
          it("should render a custom row with the custom props", () => {
            expect(
              <TestArray>
                <Row
                  {...withAllProps}
                  collapsed={true}
                  test-row-prop={Symbol("test-row-value")}
                >
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </TestArray>
            ).toMatchSnapshot();
          });
        });
      });

      describe("and the panel is not collapsed", () => {
        describe("when passed required props", () => {
          it("should render a default row with required props", () => {
            expect(
              <TestArray>
                <Row {...requiredProps} collapsed={false}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </TestArray>
            ).toMatchSnapshot();
          });
        });

        describe("when passed all props", () => {
          it("should render a custom row", () => {
            expect(
              <TestArray>
                <Row {...withAllProps} collapsed={false}>
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </TestArray>
            ).toMatchSnapshot();
          });
        });

        describe("when passed custom props", () => {
          it("should render a custom row with the custom props", () => {
            expect(
              <TestArray>
                <Row
                  {...withAllProps}
                  collapsed={false}
                  test-row-prop={Symbol("test-row-value")}
                >
                  <test-panel-1 test-panel-prop={Symbol("test-panel-value")} />
                  <test-panel-2 test-panel-prop={Symbol("test-panel-value")} />
                </Row>
              </TestArray>
            ).toMatchSnapshot();
          });
        });
      });
    });
  });
});
