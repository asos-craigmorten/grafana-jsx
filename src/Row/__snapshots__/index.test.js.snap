// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row when used in isolation when passed Panel children and the panel is collapsed when passed all props should render a custom row 1`] = `
Object {
  "collapsed": true,
  "datasource": "test-datasource",
  "gridPos": Object {
    "h": 1,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "options": Object {
    "test-option-key": "test-option-value",
  },
  "test-panels": Array [
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
  ],
  "title": "test-title",
  "type": "row",
}
`;

exports[`Row when used in isolation when passed Panel children and the panel is collapsed when passed custom props should render a custom row with the custom props 1`] = `
Object {
  "collapsed": true,
  "datasource": "test-datasource",
  "gridPos": Object {
    "h": 1,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "options": Object {
    "test-option-key": "test-option-value",
  },
  "test-panels": Array [
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
  ],
  "test-row-prop": Symbol(test-row-value),
  "title": "test-title",
  "type": "row",
}
`;

exports[`Row when used in isolation when passed Panel children and the panel is collapsed when passed required props should render a default row with required props 1`] = `
Object {
  "collapsed": true,
  "gridPos": Object {
    "h": 1,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "test-panels": Array [
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
    Object {
      "test-panel-prop": Symbol(test-panel-value),
    },
  ],
  "type": "row",
}
`;

exports[`Row when used in isolation when passed Panel children and the panel is not collapsed when passed all props should render a custom row 1`] = `
Array [
  Object {
    "collapsed": false,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "title": "test-title",
    "type": "row",
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
]
`;

exports[`Row when used in isolation when passed Panel children and the panel is not collapsed when passed custom props should render a custom row with the custom props 1`] = `
Array [
  Object {
    "collapsed": false,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
]
`;

exports[`Row when used in isolation when passed Panel children and the panel is not collapsed when passed required props should render a default row with required props 1`] = `
Array [
  Object {
    "collapsed": false,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [],
    "type": "row",
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
]
`;

exports[`Row when used in isolation when passed no Panel children when passed all props should render a custom row 1`] = `
Object {
  "collapsed": true,
  "datasource": "test-datasource",
  "gridPos": Object {
    "h": 1,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "options": Object {
    "test-option-key": "test-option-value",
  },
  "test-panels": Array [],
  "title": "test-title",
  "type": "row",
}
`;

exports[`Row when used in isolation when passed no Panel children when passed custom props should render a custom row with the custom props 1`] = `
Object {
  "collapsed": true,
  "datasource": "test-datasource",
  "gridPos": Object {
    "h": 1,
    "w": 24,
    "x": 0,
    "y": 0,
  },
  "options": Object {
    "test-option-key": "test-option-value",
  },
  "test-panels": Array [],
  "test-row-prop": Symbol(test-row-value),
  "title": "test-title",
  "type": "row",
}
`;

exports[`Row when used in isolation when passed no Panel children when passed required props should render a default row with required props 1`] = `
Array [
  Object {
    "collapsed": false,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [],
    "type": "row",
  },
]
`;

exports[`Row when used within another object when passed Panel children and the panel is collapsed when passed all props should render a custom row 1`] = `
Object {
  "panel": Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
    ],
    "title": "test-title",
    "type": "row",
  },
}
`;

exports[`Row when used within another object when passed Panel children and the panel is collapsed when passed custom props should render a custom row with the custom props 1`] = `
Object {
  "panel": Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
    ],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
}
`;

exports[`Row when used within another object when passed Panel children and the panel is collapsed when passed required props should render a default row with required props 1`] = `
Object {
  "panel": Object {
    "collapsed": true,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
    ],
    "type": "row",
  },
}
`;

exports[`Row when used within another object when passed Panel children and the panel is not collapsed when passed all props should render a custom row 1`] = `
Object {
  "panel": Object {
    "collapsed": false,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "title": "test-title",
    "type": "row",
  },
  "test-panel-1": Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  "test-panel-2": Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
}
`;

exports[`Row when used within another object when passed Panel children and the panel is not collapsed when passed custom props should render a custom row with the custom props 1`] = `
Object {
  "panel": Object {
    "collapsed": false,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
  "test-panel-1": Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  "test-panel-2": Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
}
`;

exports[`Row when used within another object when passed Panel children and the panel is not collapsed when passed required props should render a default row with required props 1`] = `
Object {
  "panel": Object {
    "collapsed": false,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [],
    "type": "row",
  },
  "test-panel-1": Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  "test-panel-2": Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
}
`;

exports[`Row when used within another object when passed no Panel children when passed all props should render a custom row 1`] = `
Object {
  "panel": Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "title": "test-title",
    "type": "row",
  },
}
`;

exports[`Row when used within another object when passed no Panel children when passed custom props should render a custom row with the custom props 1`] = `
Object {
  "panel": Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
}
`;

exports[`Row when used within another object when passed no Panel children when passed required props should render a default row with required props 1`] = `
Object {
  "panel": Object {
    "collapsed": false,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [],
    "type": "row",
  },
}
`;

exports[`Row when used within another object's array when passed Panel children and the panel is collapsed when passed all props should render a custom row 1`] = `
Array [
  Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
    ],
    "title": "test-title",
    "type": "row",
  },
]
`;

exports[`Row when used within another object's array when passed Panel children and the panel is collapsed when passed custom props should render a custom row with the custom props 1`] = `
Array [
  Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
    ],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
]
`;

exports[`Row when used within another object's array when passed Panel children and the panel is collapsed when passed required props should render a default row with required props 1`] = `
Array [
  Object {
    "collapsed": true,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
      Object {
        "test-panel-prop": Symbol(test-panel-value),
      },
    ],
    "type": "row",
  },
]
`;

exports[`Row when used within another object's array when passed Panel children and the panel is not collapsed when passed all props should render a custom row 1`] = `
Array [
  Object {
    "collapsed": false,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "title": "test-title",
    "type": "row",
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
]
`;

exports[`Row when used within another object's array when passed Panel children and the panel is not collapsed when passed custom props should render a custom row with the custom props 1`] = `
Array [
  Object {
    "collapsed": false,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
]
`;

exports[`Row when used within another object's array when passed Panel children and the panel is not collapsed when passed required props should render a default row with required props 1`] = `
Array [
  Object {
    "collapsed": false,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [],
    "type": "row",
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
  Object {
    "test-panel-prop": Symbol(test-panel-value),
  },
]
`;

exports[`Row when used within another object's array when passed no Panel children when passed all props should render a custom row 1`] = `
Array [
  Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "title": "test-title",
    "type": "row",
  },
]
`;

exports[`Row when used within another object's array when passed no Panel children when passed custom props should render a custom row with the custom props 1`] = `
Array [
  Object {
    "collapsed": true,
    "datasource": "test-datasource",
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "options": Object {
      "test-option-key": "test-option-value",
    },
    "test-panels": Array [],
    "test-row-prop": Symbol(test-row-value),
    "title": "test-title",
    "type": "row",
  },
]
`;

exports[`Row when used within another object's array when passed no Panel children when passed required props should render a default row with required props 1`] = `
Array [
  Object {
    "collapsed": false,
    "gridPos": Object {
      "h": 1,
      "w": 24,
      "x": 0,
      "y": 0,
    },
    "test-panels": Array [],
    "type": "row",
  },
]
`;
